core/src/main/scala/org/bitcoins/core/protocol/dlc/OracleInfo.scala

```diff
@@ -0,0 +1,329 @@
+  override def singleOracleInfos: Vector[EnumSingleOracleInfo]
+}
+
+/** Specifies a set of oracles for an Numeric Outcome DLC */
+sealed trait NumericOracleInfo extends OracleInfo {
+  override def singleOracleInfos: Vector[NumericSingleOracleInfo]
+}
+
+object OracleInfo
+    extends TLVDeserializable[OracleInfoTLV, OracleInfo](OracleInfoTLV) {
+
+  override def fromTLV(tlv: OracleInfoTLV): OracleInfo = {
+    tlv match {
+      case tlv: OracleInfoV0TLV => SingleOracleInfo.fromTLV(tlv)
+      case tlv: OracleInfoV1TLV => ExactMultiOracleInfo.fromTLV(tlv)
+      case tlv: OracleInfoV2TLV => NumericMultiOracleInfo.fromTLV(tlv)
+    }
+  }
+}
+
+/** Specifies a single oracles' information through an announcement */
+sealed trait SingleOracleInfo
+    extends OracleInfo
+    with TLVSerializable[OracleInfoV0TLV] {
+  override val numOracles: Int = 1
+  override val threshold: Int = 1
+
+  override def singleOracleInfos: Vector[this.type] = Vector(this)
+
+  def announcement: OracleAnnouncementTLV
+
+  /** The oracle's public key */
+  def publicKey: SchnorrPublicKey = announcement.publicKey
+
+  /** The oracle's pre-committed nonces, in the correct order */
+  def nonces: Vector[SchnorrNonce] = announcement.eventTLV.nonces
+
+  /** The order of the given sigs should correspond to the given outcome. */
+  def verifySigs(outcome: DLCOutcomeType, sigs: OracleSignatures): Boolean
+
+  /** Computes the signature point (aka signature anticipation) for a given outcome.
+    * This point is used for adaptor signing.
+    */
+  def sigPoint(outcome: DLCOutcomeType): ECPublicKey = {
+    publicKey.computeSigPoint(outcome.serialized, nonces)
+  }
+
+  /** Computes the sum of all nonces used in a given outcome */
+  def aggregateNonce(outcome: DLCOutcomeType): SchnorrNonce = {
+    nonces
+      .take(outcome.serialized.length)
+      .map(_.publicKey)
+      .reduce(_.add(_))
+      .schnorrNonce
+  }
+
+  override def toTLV: OracleInfoV0TLV = OracleInfoV0TLV(announcement)
+}
+
+object SingleOracleInfo
+    extends TLVDeserializable[OracleInfoV0TLV, SingleOracleInfo](
+      OracleInfoV0TLV) {
+
+  def apply(announcement: OracleAnnouncementTLV): SingleOracleInfo = {
+    announcement.eventTLV.eventDescriptor match {
+      case _: EnumEventDescriptorV0TLV =>
+        EnumSingleOracleInfo(announcement)
+      case _: NumericEventDescriptorTLV =>
+        NumericSingleOracleInfo(announcement)
+    }
+  }
+
+  def apply(tlv: OracleInfoV0TLV): SingleOracleInfo = {
+    SingleOracleInfo(tlv.announcement)
+  }
+
+  override def fromTLV(tlv: OracleInfoV0TLV): SingleOracleInfo = {
+    SingleOracleInfo(tlv)
+  }
+}
+
+/** Specifies a single oracles' information for an Enumerated Outcome DLC
+  * through an announcement
+  */
+case class EnumSingleOracleInfo(announcement: OracleAnnouncementTLV)
+    extends SingleOracleInfo
+    with EnumOracleInfo {
+  require(announcement.eventTLV.eventDescriptor
+            .isInstanceOf[EnumEventDescriptorV0TLV],
```

Comment:
Maybe eventually we will want to make things like `OracleAnnouncementWithEnumTLV` so we don't need these